### DISCUSSION TITLE
Change in upload metadata handling

### DISCUSSION THREAD

**Discussion by rcloran** [rcloran/lr-inaturalist-publish#5] [Polls]
I'm changing a bit about how uploads work, making it match how the web uploader works more closely by allowing the server to extract metadata from uploaded JPGs, instead of getting the metadata from Lightroom. This means that which metadata is set in the observation would match the exported photo.

That's different to how things currently work, where even if you choose not to export any metadata in the JPG (metadata panel in the settings) the plugin sets some metadata in the observation -- time, description, GPS, and keywords. I've found the Lightroom metadata  export selection a bit tricky to understand, but I don't think my current approach of forcing all the metadata is right, either.

Do you have any opinions on how this should work? I'd like to hear what you think the best way of doing it should be.

**Comment by rcloran** [comment]
To expand a bit on this...

The advantages of just using the Lightroom metadata and allowing the iNaturalist server to extract everything are that there's one way of specifying which metadata is exported, instead of having two settings. In addition, allowing both to be specified would still have the metadata uploaded as part of the JPG, so it would still be public which might be confusing. 

The advantage of allowing the plugin to fill in observation data is that you could restrict which metadata is exposed in a more fine-grained way. This makes a difference for is the timestamp, which can be excluded from the JPG metadata by accident, but doesn't seem to carry any real privacy risk. It also might matter for keywords, where setting iNaturalist tags seems to be a bit weird -- but may be what you want. 

While I voted for just using the Lightroom settings, I created this poll to clarify my thoughts on this, and I'm still not certain where I fall. If you have a use case for the other options I'm genuinely interested in hearing it.

**Reply by jjmontem** [reply]
For whatever reason, many of my image files report metadata conflicts with the Lightroom DB---I often overwrite the file metadata with that from Lightroom because I only intentionally edit stuff in Lightroom. Maybe this doesn't really apply here, but I'd be afriad of uploading a conflict from the JPG itself if iNat is reading from there.

**Reply by rcloran** [reply]
> For whatever reason, many of my image files report metadata conflicts with the Lightroom DB---I often overwrite the file metadata with that from Lightroom because I only intentionally edit stuff in Lightroom. Maybe this doesn't really apply here, but I'd be afriad of uploading a conflict from the JPG itself if iNat is reading from there.

I haven't been using LR Classic for long (only a few months, really -- I started working on this plugin soon after I moved my library over), so I'm not super familiar with all the ins and outs of how it deals with metadata, but ... I had this problem when I first imported my photos to it, too. I had sidecar xmp files with all of my raw files, and for some reason I never really understood. The only resolution I could figure out was to never have xmp files by changing the setting to write them (Catalog Settings -> Metadata -> uncheck Automatically write changes into XMP), and then deleting them all. I don't know if that's the right solution for you, but it's what worked for me 🤷‍♂️

That said, the plugin only has one way of reading the metadata, and I believe that'll come from the catalog, not what's in the file. The JPGs I'm referring to are intermediates that are created during upload. Again, they /should/ have the catalog data, as far as I know. They'd be the same files as you'd get by doing an Export -- a publish plugin is essentially just a normal export with an upload and then delete the intermediate file afterwards.
